ARTstor.js

```diff
@@ -1,131 +1,484 @@
+        dataItem.creators.push(ZU.cleanAuthor(value, contributor, false));
+    }
+}
+
+function cleanStringValue(str) {
+    var cleanValue = str.replace(/\<wbr\/>/g, "");
+    cleanValue = cleanValue.replace(/<\/?[^>]+(>|$)/g, " ");
+    return cleanValue;
+}
+
+function setItemLabelValue(doc, dataItem, key, label, value) {
+    var cleanValue = cleanStringValue(value);
+
+    if (!(key in dataItem)) {
+        dataItem[key] = label + ": " + cleanValue;
+
+    } else {
+        var fieldValue = dataItem[key];
+        if (fieldValue.indexOf(label) >= 0) {
+            dataItem[key] += ", " + cleanValue;
+        } else {
+            dataItem[key] += "; " + label + ": " + cleanValue;
+
+        }
+    }
+}
+
+function setItemValue(dataItem, key, value, override) {
+    var cleanValue = cleanStringValue(value);
+
+    if (!(key in dataItem) || override) {
+        dataItem[key] = cleanValue;
+
+    } else {
+        dataItem[key] += "; " + cleanValue;
+    }
+}
+/**
+	This procedure makes the extra call to get the notes associated
+	with the image records and add it to the Zotero data item.
+**/
+function getNotesDataItem(url, objItem, dataItem) {
+    var itemAry = objItem.split(':');
+    var objType = itemAry[1];
+    var serviceURL = getServiceUrlRoot(url) + "icommentary/" + itemAry[0];
+    Zotero.Utilities.HTTP.doGet(serviceURL,
+        function(text) {
+            var json = JSON.parse(text);
+            for (var j = 0; j < json.numberOfCommentaries; j = j + 1) {
+                if (json.ICommentary[j].status == 2) {
+                    //public commentary
+                    var comment = "";
+                    if (json.ICommentary[j].ownerName == "") {
+                        comment = "Note: ";
+                    } else {
+                        comment = "Note by: " + json.ICommentary[j].ownerName + " -  ";
+                    }
+                    comment += json.ICommentary[j].commentary;
+                    dataItem.notes.push({
+                        note: comment
+                    });
+                }
+            }
+            getResourceDataItem(url, objItem, dataItem);
+        }
+    ); //doGet
+
+}
+
+function getResourceDataItem(url, objItem, dataItem) {
+    var itemAry = objItem.split(':');
+    var serviceURL = getServiceUrlRoot(url) + "metadata/" + itemAry[0] + "/" + "?_method=FpHtml";
+
+    Zotero.Utilities.HTTP.doGet(serviceURL, function(text) {
+        var service = text.substring(text.indexOf("secure"));
+        service = service.substring(0, service.indexOf("</td>")).replace(/<wbr\/>/g, "").substring(service.indexOf("?")).trim();
+        dataItem.url = getServerUrl(url) + "/secure/ViewImages" + service + "&zoomparams=&fs=true";
+        dataItem.complete();
+    });
+}
+
+function getPortal(url) {
+    var portal = url.substring(url.indexOf('://') + 3, url.indexOf('.'));
+    return portal;
+}
+
+function getServerUrl(url) {
+    var serverUrl;
+    if (url.indexOf('/iv2\.') > 0) {
```

Comment:
is this actually what you mean? I'm guessing should be `/iv2.` or is there actually a backslash in the URL?